value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -1872,7 +1872,7 @@ checkNotIsSet([v.names.isSet](), "[v.names.raw]");
       [else]
     this.[v.name].put(
         [requireNonNull type](key, "[v.name] key"),
-        [requireNonNull type](value, "[v.name] value"));
+        [requireNonNull type](value, "[v.name] value for key: " + key));
```

Comment:
@elucash how can I check at this place whether `value` is a primitive or not? I was experimenting with 
```
[if not v.primitiveElement]value == null ? [/if][requireNonNull type](value, "[v.name] value for key: " + key)[if not v.primitiveElement] : value[/if]);
```

but it would produce not the results I was hoping for as it failed later here (since you can't do a null check without boxing):

```
public final JdkMapsBuilder putJust(long key, int value) {
    this.just.put(
        Objects.requireNonNull(key, "just key"),
        value == null ? Objects.requireNonNull(value, "just value for key: " + key) : value);
    return this;
}
```

Reply:
`v.primitiveElement` is only for keys, so it ignores values. I've found some duplicated methods(accessors) and among those there seems to be these ones which are more appropriate

* `v.unwrappedElementPrimitiveType` - if key would be primitive in `put(key, value)`, works on 1st type parameter of a map
* `v.unwrappedSecondaryElementPrimitiveType` - if value would be primitive in `put(key, value)`, works on 2nd type parameter of a map

So in this, you would just use `unwrappedSecondaryElementPrimitiveType` check

Reply:
thanks for your help, that worked.


I adjusted the condition to 
```
[if v.unwrappedSecondaryElementPrimitiveType]value[else]value == null ? [requireNonNull type](value, "[v.name] value for key: " + key) : value[/if])
```
so we're not doing a null check on a primitive and hence avoiding the cost of boxing. Let me know what you think.